sbncode/BeamSpillInfoRetriever/NuMIRetriever/NuMIRetriever_module.cc

```diff
@@ -0,0 +1,254 @@
+      try{bfp->GetNamedData(times_temps[i], "E:NSLINB",&NSLINB);}catch (WebAPIException &we) {std::cout << "At time : " << times_temps[i] << " " << "got exception: " << we.what() << "\n";}
+      try{bfp->GetNamedData(times_temps[i], "E:NSLINC",&NSLINC);}catch (WebAPIException &we) {std::cout << "At time : " << times_temps[i] << " " << "got exception: " << we.what() << "\n";}
+      try{bfp->GetNamedData(times_temps[i], "E:NSLIND",&NSLIND);}catch (WebAPIException &we) {std::cout << "At time : " << times_temps[i] << " " << "got exception: " << we.what() << "\n";}
+      // BPM Positions and Intensities - they each have 7 elements
+      // First is an average value of a few batches (often 2,3,4)
+      // used for auto-tuning, so we should disregard it
+      
+      try{HP121 = bfp->GetNamedVector(times_temps[i], "E:HP121[]");}catch (WebAPIException &we) {std::cout << "At time : " << times_temps[i] << " " << "got exception: " << we.what() << "\n";}
+      try{VP121 = bfp->GetNamedVector(times_temps[i], "E:VP121[]");}catch (WebAPIException &we) {std::cout << "At time : " << times_temps[i] << " " <<"got exception: " << we.what() << "\n";}
+      try{HPTGT = bfp->GetNamedVector(times_temps[i], "E:HPTGT");}catch (WebAPIException &we) {std::cout << "At time : " << times_temps[i] << " " <<"got exception: " << we.what() << "\n";}
+      try{VPTGT = bfp->GetNamedVector(times_temps[i], "E:VPTGT");}catch (WebAPIException &we) {std::cout << "At time : " << times_temps[i] << " " <<"got exception: " << we.what() << "\n";}
+      try{HITGT = bfp->GetNamedVector(times_temps[i], "E:HITGT");}catch (WebAPIException &we) {std::cout << "At time : " << times_temps[i] << " " <<"got exception: " << we.what() << "\n";}
+      try{VITGT = bfp->GetNamedVector(times_temps[i], "E:VITGT");}catch (WebAPIException &we) {std::cout << "At time : " << times_temps[i] << " " <<"got exception: " << we.what() << "\n";}
+      
+      std::cout << "Finished getting IFBeam info" << std::endl;
+      std::cout << "BFP Time: " << times_temps[i] << " TOROID Time: " << TRTGTD_time << std::endl;
+      unsigned long int time_closest_int = (int) TRTGTD_time;
+      double time_closest_ns = (TRTGTD_time - time_closest_int)*1000000000;
+
+      sbn::NuMISpillInfo NuMIbeamInfo;
+      NuMIbeamInfo.TRTGTD = TRTGTD;
+      NuMIbeamInfo.TR101D = TR101D;
+      NuMIbeamInfo.HRNDIR = HRNDIR;
+      NuMIbeamInfo.NSLINA = NSLINA;
+      NuMIbeamInfo.NSLINB = NSLINB;
+      NuMIbeamInfo.NSLINC = NSLINC;
+      NuMIbeamInfo.NSLIND = NSLIND;
+      
+      NuMIbeamInfo.HP121 = HP121;
+      NuMIbeamInfo.VP121 = VP121;
+      NuMIbeamInfo.HPTGT = HPTGT;
+      NuMIbeamInfo.VPTGT = VPTGT;
+      NuMIbeamInfo.HITGT = HITGT;
+      NuMIbeamInfo.VITGT = VITGT;
+
+      NuMIbeamInfo.event = e.event();
+      NuMIbeamInfo.spill_time_s = time_closest_int;
+      NuMIbeamInfo.spill_time_ns = time_closest_ns;
+      // Save the Number of DAQ Gates in the first saved spill
+      if (spill_count == 1) {
```

Comment:
What is this bit of code supposed to be doing?

Reply:
Do you mean the spill count thing? 

Since the number of DAQ gates seems like an important number for (e.g.) quality checks, I also wanted to save that information. The # of DAQ gates is per-event, while the rest of the NuMI information is per-NuMI-spill. So I decided to save the number of DAQ gates in the first NuMI spill per event.